src/Injector/AbstractInjector.php

```diff
@@ -162,6 +181,155 @@ public function inject($package, $type, IOInterface $io)
     }
 
     /**
+     * Inject component $package after all dependencies into $config and return true.
+     * If any of dependencies is not registered the method will write error to $io
+     * and will also return true, to prevent injecting this package later.
+     * Method return false only in case when dependencies for the package are not found.
+     *
+     * @param string $package
+     * @param string $config
+     * @param IOInterface $io
+     * @return bool
+     */
+    private function injectAfterDependencies($package, $config, IOInterface $io)
+    {
+        $moduleClass = sprintf('%s\\%s', $package, 'Module');
+        if (class_exists($moduleClass) && method_exists($moduleClass, 'getModuleDependencies')) {
+            $module = new $moduleClass();
+            $dependencies = $module->getModuleDependencies();
+
+            if ($dependencies) {
+                foreach ($dependencies as $dependency) {
+                    if (! $this->isRegisteredInConfig($dependency, $config)) {
+                        $io->write(sprintf(
+                            '<error>    Dependency %s is not registered in the configuration</error>',
+                            $dependency
+                        ));
+
+                        return true;
+                    }
+                }
+
+                $lastDependency = $this->findLastDependency($dependencies, $config);
+
+                $pattern = sprintf(
+                    $this->injectionPatterns[self::TYPE_DEPENDENCY]['pattern'],
+                    preg_quote($lastDependency, '/')
+                );
+                $replacement = sprintf(
+                    $this->injectionPatterns[self::TYPE_DEPENDENCY]['replacement'],
+                    $package
+                );
+
+                $config = preg_replace($pattern, $replacement, $config);
+                file_put_contents($this->configFile, $config);
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Find which of dependency packages is the last one on the module list.
+     *
+     * @param array $dependencies
+     * @param string $config
+     * @return string
+     */
+    private function findLastDependency(array $dependencies, $config)
+    {
+        if (count($dependencies) == 1) {
+            return reset($dependencies);
```

Comment:
This is fine, before we use this method we check if all dependencies are registered, but again, this is a bit tricky, and the function is not completely independent.